features/collection/api/src/main/java/org/opennms/netmgt/collection/api/ServiceCollectorRegistry.java

```diff
@@ -37,7 +38,7 @@
  */
 public interface ServiceCollectorRegistry {
 
-    ServiceCollector getCollectorByClassName(String className);
+    CompletableFuture<ServiceCollector> getCollectorByClassName(String className);
```

Comment:
Seeing how this is used in other places, I think we should create a new method, rather than changing the current one.

Maybe `CompletableFuture<ServiceCollector> getCollectorFutureByClassName(String className)`?

The existing method could then have the previous behaviour of returning `null` if it does not currently exist in the registry.